### PR TITLE
MeTTa and Cypher Annotation Support

### PR DESCRIPTION
**Pull request by eyasubirhanu**

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -26,32 +27,29 @@ def process_query():
...
+
     try:
-        db_instance = databases[database_type]
         requests = data['requests']
-        query_code = db_instance.query_Generator(requests)
+        
+        # Validate the request data before processing
+        node_map = validate_request(requests, schema_manager.schema)
+        
+        database_type = 'cypher'
```
Comment (tanksha) [inline app/routes.py:37]: I suggest you use a config file to specify which database type to use, that way it is easy to alternate between Metta and cypher

File: app/services/cypher_generator.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+    def run_query(self, query_code):
+        with self.driver.session() as session:
+            results = session.run(query_code[0])
+            result_list = [record for record in results]
+            return result_list
+
+    def query_Generator(self, requests,node_map):
+        if node_map is None:
+            raise Exception('error')
```
Comment (tanksha) [inline app/services/cypher_generator.py:70]: Please add more information on what error is being raised here

File: app/services/metta_generator.py [hunk 3]
```diff
@@ -43,89 +43,209 @@ def construct_node_representation(self, node, identifier):
             node_representation += f' ({key} ({node_type + " " + identifier}) {value})'
         return node_representation
 
-    def query_Generator(self, requests):
+    def query_Generator(self, data,node_map):
+
+        if node_map is None:
+            raise Exception('error')
```
Comment (tanksha) [inline app/services/metta_generator.py:49]: here as well, more information about the error

File: run.py [hunk 4]
```diff
@@ -2,3 +2,4 @@
 
 if __name__ == '__main__':
     app.run(debug=True, host='0.0.0.0', port=5000)
+
```
Comment (tanksha) [inline run.py:5]: add a newline here

**Review by tanksha — CHANGES_REQUESTED** [review]
This looks good. Please fix the comments I left and I will merge it.

File: app/routes.py (hunk 1 above)
Comment (eyasubirhanu) [reply app/routes.py]: I’ve added a config file

File: app/services/cypher_generator.py (hunk 2 above)
Comment (eyasubirhanu) [reply app/services/cypher_generator.py]: added more details about the specific error

File: run.py (hunk 4 above)
Comment (eyasubirhanu) [reply run.py:6]: added newline

**Comment by tanksha** [comment]
Thanks. merging it.

**Pull request merged**